internal/cmd/test/build_cmd.go

```diff
@@ -0,0 +1,112 @@
+package test
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/gobuffalo/envy"
+	"github.com/gobuffalo/meta"
+)
+
+var (
+	// if GO_BIN is set, use it, otherwise use "go"
+	goBinary = envy.Get("GO_BIN", "go")
+
+	// testValuedFlags are the flags on the go test command that
+	// receive a value, we use this to determine if the last argument
+	// is a flag or a package path
+	testValuedFlags = strings.Join([]string{
+		`-p`, `-asmflags`, `-buildmode`, `-compiler`, `-gccgoflags`,
+		`-gcflags`, `-installsuffix`, `-mod`, `-modfile`, `-overlay`,
+		`-pkgdir`, `-tags`, `-trimpath`, `-toolexec`, `-o`, `-exec`,
+		`-bench`, `-benchtime`, `-blockprofile`, `-blockprofilerate`,
+		`-count`, `-coverprofile`, `-cpu`, `-cpuprofile`, `-fuzz `,
+		`-fuzzcachedir`, `-fuzzminimizetime`, `-fuzztime`, `-list`,
+		`-memprofile`, `-memprofilerate`, `-mutexprofile`, `-mutexprofilefraction`,
+		`-outputdir`, `-parallel`, `-run`, `-shuffle`, `-testlogfile`,
+		`-timeout`, `-trace`,
+	}, "|")
+)
+
+// buildCmd builds the test command to be passed to the go test command
+// after cutting some of the arguments that are specific to buffalo and adding
+// packages if missing.
+func buildCmd(args []string) (*exec.Cmd, error) {
+	app := meta.New(".")
+
+	ccar := clean(args)
+	cargs := append([]string{
+		"test",
+		// run sequentially
+		"-p", "1",
+		// add build tags
+		"-tags", app.BuildTags("development").String(),
+		//TODO: Should we merge it with passed tags?
+	}, ccar...)
+
+	// if no packages are specified, add the current directory
+	lastIsFlag := len(ccar) > 0 && strings.HasPrefix(args[len(ccar)-1], "-")
+	lastIsFlagValue := len(ccar) >= 2 && strings.Contains(testValuedFlags, ccar[len(ccar)-2])
+	if len(ccar) == 0 || lastIsFlag || lastIsFlagValue {
+		pkgs, err := findPackages()
+		if err != nil {
+			return nil, err
+		}
+
+		cargs = append(cargs, pkgs...)
+	}
+
+	// Add extra args (-testify.m) to the command
+	cargs = append(cargs, extraArgs(args)...)
```

Comment:
I found that the function `clean()` (only) removes `-m` and `-testify.m` from the args and the function `extraArgs()` appends `-testify.m` at the end of the command line after the list of packages when one of `-m`, `-testify.m`, or `-run` was specified. However, it will not work if the user provides `-run TestMethod` since when a user wants to run a specific test case in a single Suite, the proper command line should be something like `go test -run TestSuite -testify.m TestMethod`. (see another comment for the old code)

Also, `-run` does not always mean to be `-testify.m`. E.g. I have a package named `utils` in my app which does not use testify (suite) based test cases which to be run with the standard `-run TestUtilsFunc`.

Reply:
Yes. The biggest issue I'm facing is maintaining a functionality that I don't see documented by our tests.

Reply:
Right, we just mentioned the feature in short words, we just supported a simple use case only, and that is for the season of v0.10, which is the season around Go v1.7 or v1.8 maybe. I think we could adapt the testing to the modern version of Go and Buffalo.

https://gobuffalo.io/documentation/guides/testing/#execute-a-single-test

Reply:
Yes. One consideration to bring here is that the -m feature can be addressed with `-run`, developers just have to enter a larger name for the tests. In general I would love to see Buffalo being closer to the sdlib for our upcomming versions.

Reply:
Agreed. I would like to make it easy for Go developers who already know how `go` works.